tests/misc.js

```diff
@@ -478,3 +478,531 @@ describe('literal JSON', () => {
+        },
+        'id': "urn:absoluteiri",
+        '@type': ["relativeiri", "definedTerm" ],
+        definedTerm: "is defined"
+      };
+
+      let expansionMapCalled = false;
+      const expansionMap = info => {
+        if(info.relativeIri === 'relativeiri') {
+          expansionMapCalled = true;
+        }
+      };
+
+      await jsonld.expand(docWithRelativeIriId, {expansionMap});
+
+      assert.equal(expansionMapCalled, true);
+    });
+
+    it('should be called on relative iri for aliased type term', async () => {
+      const docWithRelativeIriId = {
+        '@context': {
+          'type': "@type",
+          'definedTerm': 'https://example.com#definedTerm'
+        },
+        'id': "urn:absoluteiri",
+        'type': "relativeiri",
+        definedTerm: "is defined"
+      };
+
+      let expansionMapCalled = false;
+      const expansionMap = info => {
+        if(info.relativeIri === 'relativeiri') {
+          expansionMapCalled = true;
+        }
+      };
+
+      await jsonld.expand(docWithRelativeIriId, {expansionMap});
+
+      assert.equal(expansionMapCalled, true);
+    });
+
+    it("should be called on relative iri when \
+    @base value is './'", async () => {
+      const docWithRelativeIriId = {
+        '@context': {
+          "@base": "./",
+        },
+        '@id': "relativeiri",
+      };
+
+      let expansionMapCalled = false;
+      const expansionMap = info => {
+        if(info.relativeIri === '/relativeiri') {
+          expansionMapCalled = true;
+        }
+      };
+
+      await jsonld.expand(docWithRelativeIriId, {expansionMap});
+
+      assert.equal(expansionMapCalled, true);
+    });
+
+    it("should be called on relative iri when \
+    @base value is './'", async () => {
+      const docWithRelativeIriId = {
+        '@context': {
+          "@base": "./",
+        },
+        '@id': "relativeiri",
+      };
+
+      let expansionMapCalled = false;
+      const expansionMap = info => {
+        if(info.relativeIri === '/relativeiri') {
+          expansionMapCalled = true;
+        }
+      };
+
+      await jsonld.expand(docWithRelativeIriId, {expansionMap});
+
+      assert.equal(expansionMapCalled, true);
+    });
+
+    it("should be called on relative iri when \
+    @vocab value is './'", async () => {
+      const docWithRelativeIriId = {
+        '@context': {
+          "@vocab": "./",
+        },
+        '@type': "relativeiri",
+      };
+
+      let expansionMapCalled = false;
+      const expansionMap = info => {
+        if(info.relativeIri === '/relativeiri') {
+          expansionMapCalled = true;
+        }
+      };
+
+      await jsonld.expand(docWithRelativeIriId, {expansionMap});
+
+      assert.equal(expansionMapCalled, true);
+    });
+  });
+
+  describe('prependedIri', () => {
+    it("should be called when property is \
+    being expanded with `@vocab`", async () => {
+      const doc = {
+        '@context': {
+          "@vocab": "http://example.com/",
```

Comment:
was there a release that included this?

Reply:
@OR13 -- a release was made with the new `safe` mode that does a more low-level and targeted approach to solving the "undefined term / relative URL" problem with a much simpler API (set `safe: true` in the options). The latest version of jsonld.js (v8) now sets `safe: true` by default in `canonize` so that method will fail closed if you don't define your terms or use absolute URLs. You can override the default settings of course, as usual.